File: app/src/main/java/com/appbusters/robinkamboj/senseitall/SensorApplication.java
package com.appbusters.robinkamboj.senseitall;

import android.app.Application;

import io.github.inflationx.calligraphy3.CalligraphyConfig;
import io.github.inflationx.calligraphy3.CalligraphyInterceptor;
import io.github.inflationx.viewpump.ViewPump;

/**
 * Created by robinkamboj on 25/01/18.
 */

public class SensorApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        ViewPump.init(ViewPump.builder()
                .addInterceptor(new CalligraphyInterceptor(
                        new CalligraphyConfig.Builder()
                                .setDefaultFontPath("fonts/spb.otf")
                                .setFontAttrId(R.attr.fontPath)
                                .build()))
                .build());
    }

}

File: app/src/main/java/com/appbusters/robinkamboj/senseitall/SensorApplication.kt
package com.appbusters.robinkamboj.senseitall

import android.app.Application
import com.appbusters.robinkamboj.senseitall.di.component.DaggerSiaApplicationComponent
import com.appbusters.robinkamboj.senseitall.di.component.SiaApplicationComponent
import com.appbusters.robinkamboj.senseitall.di.module.ContextModule

import io.github.inflationx.viewpump.ViewPump
import javax.inject.Inject

class SensorApplication : Application() {

    lateinit var viewPump: ViewPump

    private lateinit var applicationComponent: SiaApplicationComponent

    override fun onCreate() {
        super.onCreate()

        applicationComponent = DaggerSiaApplicationComponent.builder()
                .contextModule(ContextModule(this))
                .build()

        viewPump = applicationComponent.viewPump()

        ViewPump.init(viewPump)
    }

    fun getApplicationComponent(): SiaApplicationComponent {
        return applicationComponent
    }
}
